### PR TITLE
Describe connection problems/HTTP errors when validating images

### PR DESCRIPTION
**Pull request by suve**
This changeset makes the image validation code display some descriptive error messages when downloading the image fails. Instead of a simple _"url not found"_ message, the user either gets a _"failed to connect message"_ or an HTTP error code. This should make it easier for users to differentiate between faulty networks, unresponsive or badly configured servers.

The `soup_status_get_phrase()` function is available since `libsoup-2.4`. ([link](https://git.gnome.org/browse/libsoup/tree/libsoup/soup-status.h#n96))

This fixes issue #200.

### CODE REVIEW

File: libappstream-glib/as-app-validate.c [hunk 1]
```diff
@@ -471,10 +472,15 @@ ai_app_validate_image_check (AsImage *im, AsAppValidateHelper *helper)
...
+	if (SOUP_STATUS_IS_TRANSPORT_ERROR(status_code)) {
 		ai_app_validate_add (helper,
 				     AS_PROBLEM_KIND_URL_NOT_FOUND,
-				     "<screenshot> url not found [%s]", url);
+				     "<screenshot> failed to connect: %s [%s] ", soup_status_get_phrase(status_code), url);
+		return FALSE;
+	} else if (status_code != SOUP_STATUS_OK) {
+		ai_app_validate_add (helper,
+				     AS_PROBLEM_KIND_URL_NOT_FOUND,
+				     "<screenshot> HTTP %d: %s [%s]", status_code, soup_status_get_phrase(status_code), url);
```
Comment (hughsie) [inline libappstream-glib/as-app-validate.c]: maybe something like `<screenshot> failed to download, HTTP error was %s [%d]`

Also, both added lines need wrapping to 80ish chars wide. Thanks.

**Comment by suve** [comment]
To fit in the 80-char limit, I changed the indentation a bit. I wonder if the first argument to `ai_app_validate_add()` should't now go on a separate line, to be more consistent with the rest of the arguments.

**Comment by hughsie** [comment]
If you merge that into one commit and force push to update the PR I think it's good to merge. Thanks.

**Comment by suve** [comment]
So a squash, then? Sure.

**Comment by suve** [comment]
Done.

**Pull request merged**